metrics_utility/automation_controller_billing/extract/base.py

```diff
@@ -101,6 +100,15 @@ def build_data_batch(self, temp_dir, file_name):
 
         if os.path.exists(os.path.join(temp_dir, f'{file_name}.csv')):
             return pd.read_csv(os.path.join(temp_dir, f'{file_name}.csv'))
+        else:
+            self.default_column_headers(temp_dir, file_name)
+
+    @staticmethod
+    def default_column_headers(temp_dir, file_name):
+        # if file_name is 'main_indirectmanagednodeaudit':
+        if os.path.exists(os.path.join(temp_dir, f'{file_name}.csv')):
+            row1 = pd.read_csv(os.path.join(temp_dir, f'{file_name}.csv'), nrows=0)
+            return pd.DataFrame(columns=row1.columns)
         else:
             return pd.DataFrame([{}])
```

Comment:
The extractor is called by the specific dataframe class - so this is not needed, the dataframe can always pass in its own `data_columns`